code/datums/elements/climbable.dm

```diff
@@ -0,0 +1,104 @@
+/datum/element/climbable
+	element_flags = ELEMENT_BESPOKE|ELEMENT_DETACH
+	id_arg_index = 2
+	///Time it takes to climb onto the object
+	var/climb_time = 20
+	///Stun duration for when you get onto the object
+	var/climb_stun = 20
+	///Assoc list of object being climbed on - climbers.  This allows us to check who needs to be shoved off a climbable object when its clicked on.
+	var/list/current_climbers
+
+/datum/element/climbable/Attach(datum/target, climb_time, climb_stun)
+	. = ..()
+
+	if(!isatom(target) || isarea(target))
+		return ELEMENT_INCOMPATIBLE
+	if(climb_time)
+		src.climb_time = climb_time
+	if(climb_stun)
+		src.climb_stun = climb_stun
+
+	RegisterSignal(target, COMSIG_ATOM_ATTACK_HAND, .proc/attack_hand)
+	RegisterSignal(target, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+	RegisterSignal(target, COMSIG_MOUSEDROPPED_ONTO, .proc/mousedrop_receive)
+	RegisterSignal(target, COMSIG_ATOM_BUMPED, .proc/try_speedrun)
+
+/datum/element/climbable/Detach(datum/target)
+	UnregisterSignal(target, list(COMSIG_ATOM_ATTACK_HAND, COMSIG_PARENT_EXAMINE, COMSIG_MOUSEDROPPED_ONTO, COMSIG_ATOM_BUMPED))
+	return ..()
+
+/datum/element/climbable/proc/on_examine(atom/source, mob/user, list/examine_texts)
+	SIGNAL_HANDLER
+
+	if(can_climb(source, user))
+		examine_texts += "<span class='notice'>[source] looks climbable</span>"
+
+/datum/element/climbable/proc/can_climb(atom/source, mob/user)
```

Comment:
Is this overridden at any point? Or is this just for future proofing?

Reply:
its just encapsulation really, I dislike jamming everything into one proc